pkg/grpc/client.go

```diff
@@ -144,3 +146,46 @@ func grpcTimeoutInterceptor(timeout time.Duration) grpc.UnaryClientInterceptor {
 		return invoker(ctx, method, req, reply, cc, opts...)
 	}
 }
+
+type grpcClientConnRecord struct {
+	conn *grpc.ClientConn
+	opts *Options
+}
+
+var grpcClientConns = struct {
+	sync.Mutex
+	m map[string]grpcClientConnRecord
+}{
+	m: make(map[string]grpcClientConnRecord),
+}
+
+// GetGRPCClientConn returns a gRPC client connection for the given name. If a connection for that name has already been
+// established the existing connection will be returned. If any options change for that connection, the existing
+// connection will be closed and a new one established.
+func GetGRPCClientConn(name string, opts *Options) (*grpc.ClientConn, error) {
+	grpcClientConns.Lock()
+	defer grpcClientConns.Unlock()
+
+	current, ok := grpcClientConns.m[name]
+	if ok {
+		if cmp.Equal(current.opts, opts) {
```

Comment:
Could we add test for `GetGRPCClientConn`.

Reply:
> equality is determined by recursively comparing the primitive kinds on both values, much like reflect.DeepEqual. Unlike reflect.DeepEqual, unexpected fields are not compared by default;

Mostly a note for myself, but options does not have any unexported fields so this is better.